Description: better fullscreen
 update fullscreen function, to make go fullscreen and back, with the same key
Author: Nicolas HOUDELOT <nicolas@demosdebs.org>
Applied-Upstream: no
Last-Update: 2020-04-28
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/main.c
+++ b/main.c
@@ -88,6 +88,13 @@
     SDL_RenderSetViewport(sdlRenderer, &newWindowSize);
 }
 
+void ToggleFullscreen(SDL_Window* Window) {
+    int FullscreenFlag = SDL_WINDOW_FULLSCREEN_DESKTOP;
+    int isFullscreen = SDL_GetWindowFlags(Window) & FullscreenFlag;
+    SDL_SetWindowFullscreen(Window, isFullscreen ? 0 : FullscreenFlag);
+    SDL_ShowCursor(isFullscreen);
+}
+
 int main(int argc,char **argv)
 {
 	int modpos,pt_songpos,pt_patternpos,quit=0,frame=0,n,i;
@@ -181,6 +188,7 @@
 				switch (e.key.keysym.sym) {
 					case SDLK_ESCAPE: quit=1;; break;
 					case SDLK_q: quit=1;; break;
+					case SDLK_f: ToggleFullscreen(sdlWindow); break;
 				}
 			}
 			if (e.window.windowID == windowID)  {
